src/transposition.c

```diff
@@ -87,6 +87,32 @@ void clearTT() {
     memset(Table.buckets, 0, sizeof(TTBucket) * (Table.hashMask + 1u));
 }
 
+void lockTTBucket(TTBucket *bucket) {
+
+#ifndef NO_TT_LOCKS
+    uint8_t previous;
+
+    do {
+        previous =
+            atomic_exchange_explicit(
+                &bucket->lock,
+                1, // desired
+                memory_order_acquire);
+    } while (previous == 1);
```

Comment:
Note: This should really have a back-off mechanism (i.e., increasing delay per iterations) in case we didn't get the lock. Hammering the lock isn't actually that great for overall perf in case there's actual contention buckets.

But let's discuss first whether you even want this code.